### PR TITLE
{drivers/apds99xx, sys/net/gnrc/netreg}: Drop xtimer includes

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

tl;dr: I get compilation issues with `xtimer.h` is included, but `USEMODULE += xtimer` is missing when the clock config of the `wasmote-pro` is fixed. See https://github.com/RIOT-OS/RIOT/pull/14799 for details on why this is so.

This PR drops includes to `xtimer.h` where the module is not actually used.

- drivers/apds99xx is not using xtimer, but including `xtimer.h`. This resulted in compilation issues on the `waspmote-pro`
    - `xtimer.h` pulled in `mutex.h` indirectly. I had to explicitly included it now to fix compilation issues
- sys/net/gnrc/netreg is including `xtimer.h` indirectly through `net/gnrc/tcp.h`, even when `gnrc_tcp` is not used
    - Stop including `net/gnrc/tcp.h` unless module `gnrc_tcp` is used
- sys/net/gnrc/netif/gnrc_netif.c includes `xtimer.h`, but only needs xtimer when `CONFIG_GNRC_NETIF_MIN_WAIT_AFTER_SEND_US` is set to a positive number (default is zero)
    - Btw: Is `xtimer` pulled in as dependency by the build system, if it is non-zero. (But out of scope of this PR.)

### Testing procedure

This should change no binaries.

### Issues/PRs references

Spun out of https://github.com/RIOT-OS/RIOT/pull/14799

### CODE REVIEW

File: sys/net/gnrc/netif/gnrc_netif.c [hunk 1]
```diff
@@ -34,7 +34,9 @@
 #include "fmt.h"
 #include "log.h"
 #include "sched.h"
+#if (CONFIG_GNRC_NETIF_MIN_WAIT_AFTER_SEND_US > 0U)
 #include "xtimer.h"
+#endif
```
Comment (maribu) [inline sys/net/gnrc/netif/gnrc_netif.c:39]: Ah, yes. Here is well. Forgot this in the PR description

Comment (maribu) [reply sys/net/gnrc/netif/gnrc_netif.c:39]: Updated PR description accordingly

**Review by benpicco — APPROVED** [review]
Murdock is happy.

**Comment by maribu** [comment]
Thanks :-)
